build/Build.cs

```diff
@@ -0,0 +1,195 @@
+using System;
+using System.Linq;
+using Nuke.Common;
+using Nuke.Common.CI;
+using Nuke.Common.Execution;
+using Nuke.Common.Git;
+using Nuke.Common.IO;
+using Nuke.Common.ProjectModel;
+using Nuke.Common.Tooling;
+using Nuke.Common.Tools.DotNet;
+using Nuke.Common.Tools.GitVersion;
+using Nuke.Common.Utilities.Collections;
+using static Nuke.Common.EnvironmentInfo;
+using static Nuke.Common.IO.FileSystemTasks;
+using static Nuke.Common.IO.PathConstruction;
+using static Nuke.Common.Tools.DotNet.DotNetTasks;
+using static Nuke.Common.Tools.DocFX.DocFXTasks;
+using System.Text.Json;
+using System.IO;
+using static Nuke.Common.Tools.NuGet.NuGetTasks;
+using static Nuke.Common.ChangeLog.ChangelogTasks;
+using static Nuke.Common.Tools.Git.GitTasks;
+using Nuke.Common.ChangeLog;
+using System.Collections.Generic;
+using Nuke.Common.Tools.GitHub;
+using Nuke.Common.Utilities;
+
+[CheckBuildProjectConfigurations]
+[ShutdownDotNetAfterServerBuild]
+partial class Build : NukeBuild
+{
+    /// Support plugins are available for:
+    ///   - JetBrains ReSharper        https://nuke.build/resharper
+    ///   - JetBrains Rider            https://nuke.build/rider
+    ///   - Microsoft VisualStudio     https://nuke.build/visualstudio
+    ///   - Microsoft VSCode           https://nuke.build/vscode
+
+    public static int Main() => Execute<Build>(x => x.CreatePackage);
+
+    [Parameter("Configuration to build - Default is 'Debug' (local) or 'Release' (server)")]
+    readonly Configuration Configuration = IsLocalBuild ? Configuration.Debug : Configuration.Release;
+
+    [Solution] readonly Solution Solution;
+    [GitRepository] readonly GitRepository GitRepository;
+    [GitVersion] readonly GitVersion GitVersion;
+
+    [Parameter] string NugetPrerelease;
+
+    // Directories
+    AbsolutePath ToolsDir => RootDirectory / "tools";
+    AbsolutePath Output => RootDirectory / "bin";
+    AbsolutePath OutputNuget => Output / "nuget";
+    AbsolutePath OutputTests => RootDirectory / "TestResults";
+    AbsolutePath OutputPerfTests => RootDirectory / "PerfResults";
+    AbsolutePath SourceDirectory => RootDirectory / "src";
+    AbsolutePath DocSiteDirectory => RootDirectory / "docs/_site";
+    AbsolutePath ArtifactsDirectory => RootDirectory / "artifacts";
+    public string ChangelogFile => RootDirectory / "CHANGELOG.md";
+
+    [Parameter] readonly string Source = "https://resharper-plugins.jetbrains.com/api/v2/package";
+
+    static readonly JsonElement? _githubContext = string.IsNullOrWhiteSpace(EnvironmentInfo.GetVariable<string>("GITHUB_CONTEXT")) ? 
+        null 
+        : JsonSerializer.Deserialize<JsonElement>(EnvironmentInfo.GetVariable<string>("GITHUB_CONTEXT"));
+
+    //let hasTeamCity = (not (buildNumber = "0")) // check if we have the TeamCity environment variable for build # set
+    static readonly int BuildNumber = _githubContext.HasValue ? int.Parse(_githubContext.Value.GetProperty("run_number").GetString()) : 0;
```

Comment:
The reason for enabling `GITHUB_CONTEXT` is to have access to `run_number` and lot more!